Rework Gateway API, HTTPRoute, and GRPCRoute docs

wmorgan opened the pull request:
Here's my attempt at reflecting the modern world of Linkerd. (Note: these are all in the 2.17/ subfolder; once we're good on content we should also just copy these over to 2-edge.)

Summary: 
- Remove the top-level HTTPRoute feature doc. Replace it with a "Gateway API" feature doc
  - This doc includes instructions for how to have another project own the GW API on your cluster
  - This doc includes the version compatibility matrix
- Remove all language suggesting that users use the `policy.linkerd.io` types, and instead note that they are supported but deprecated
- Ensure all ServiceProfile docs (feature, ref, and guide) have a note that GW API types are now preferred, at the top
- Add a new GRPCRoute reference doc
- Make the HTTPRoute and (new) GRPCRoute reference docs point to the corresponding official docs for these types, and describe these docs as supplementary

Please review for technical accuracy as well as completeness. Note that the goal is to provide a practical guide to using the GW API types with Linkerd, not a comprehensive treatment of the GW API. Feedback welcome

kflynn commented on linkerd.io/content/2.17/tasks/restricting-access.md:
```diff
@@ -167,16 +167,18 @@ explicitly create an authorization to allow those probe requests. For more
 You may have noticed that there was a period of time after we created the
 `Server` resource but before we created the `ServerAuthorization` where all
 requests were being rejected. To avoid this situation in live systems, we
-recommend that you enable [audit mode](../../features/server-policy/#audit-mode)
-in the `Server` resource (via `accessPolicy:audit`) and check the proxy
-logs/metrics in the target services to see if traffic would get inadvertently
-denied. Afterwards, when you're sure about your policy rules, you can fully
-enable them by resetting `accessPolicy` back to `deny`.
+recommend that you start with [audit
+mode](../../features/server-policy/#audit-mode) enabled on the `Server`
+resource. In this mode, traffic that violates the policy will not actually be
+denied, and you will be able to check the proxy logs/metrics on the target
+services for a complete picture of what would happen when audit mode is
+disabled. Once you're sure about your policy rules, you can fully enable them by
+removing audit mode to enforce the policies.
```
One request: @wmorgan, can you pull this change out of this PR? It has nothing to do with Gateway API, and may not be complete anyway...

wmorgan replied:
This was another change I was planning on making so I'd rather leave it in unless you object to the content? At the risk of a messier PR scope.

kflynn replied:
The content is fine, it's just the scope thing...

wmorgan commented:
I fixed the broken links. In doing that I also cleaned up those docs, which expanded the scope of this PR quite a bit. We had a couple "tasks" docs that were very low-effort, which I removed entirely and put the content in the feature doc. I cleaned up a few other stylistic things like excessive hyperlinking.

kflynn reviewed (approved):
I like this so far, many thanks!! 🙂

wmorgan commented:
Ok, I think I have finally made the linters happy. @travisbeckham should I also copy these changes over to /2-edge/ ? I think they are also applicable to 2.16, should I copy them to that subtree as well?
